Asterix/optics/phase_amplitude_functions.py

```diff
@@ -220,3 +221,410 @@ def sine_cosine_basis(Nact1D):
+            center_on_pixel = False
+        elif center_pos.lower() == 'p':
+            dimpp_pup_large = (2 * int(np.ceil(prad)) + 1) * grey_pup_bin_factor
+            center_on_pixel = True
+        else:
+            raise ValueError("center_pos must be 'p' (centered on pixel) or 'b' (centered in between 4 pixels)")
+
+        apodizer_pupil_large = make_apodizer(dimpp_pup_large,
+                                             grey_pup_bin_factor * prad,
+                                             apodizer_profile,
+                                             grey_pup_bin_factor=1,
+                                             center_pos=center_pos)
+        return crop_or_pad_image(
+            rebin(apodizer_pupil_large, factor=grey_pup_bin_factor, center_on_pixel=center_on_pixel), dim_pp)
+
+    else:
+        xx, yy = np.meshgrid(np.arange(dim_pp) - dim_pp // 2, np.arange(dim_pp) - dim_pp // 2)
+
+        if center_pos.lower() == 'b':
+            xx = xx + 1 / 2
+            yy = yy + 1 / 2
+        elif center_pos.lower() == 'p':
+            pass
+        else:
+            raise ValueError("center_pos must be 'p' (centered on pixel) or 'b' (centered in between 4 pixels)")
+        rr = np.hypot(yy, xx)
+
+        apodizer_pupil = np.zeros((dim_pp, dim_pp))
+        apodizer_pupil[rr <= prad] = apodizer_profile(rr[rr <= prad] / prad)
+
+        return apodizer_pupil
+
+
+def make_spider(dim_pp, starting_point, finishing_point, w_spiders, center_pos='b'):
+    """
+    Make a unique spider from starting_point to finishing point, of width w_spiders
+
+    AUTHORS : Johan Mazoyer, heavily inspired by Emiel Por in HCIpy
+
+    Parameters
+    ----------
+    dim_pp : int
+        Size of the pupil plane (in pixels)
+    starting_point: tupple of float
+        (x0,y0) the starting point of the spider in pixel
+    finishing_point: tupple of float
+        (x0,y0) the finishing point of the spider in pixel
+    w_spiders: float
+        width of the spider in pixel
+    center_pos : string (optional, default 'b')
+        Option for the center pixel.
+        If 'p', center on the pixel dim_pp//2.
+        If 'b', center in between pixels dim_pp//2 -1 and dim_pp//2, for 'dim_pp' odd or even.
+
+    Returns
+    ------
+    spider_map : 2D bool array
+        spider boolean array
+    """
+
+    delta = np.array(finishing_point) - np.array(starting_point)
+    shift = delta / 2 + np.array(starting_point)
+
+    spider_angle = np.arctan2(delta[1], delta[0])
+    spider_length = np.sqrt(delta[0]**2 + delta[1]**2)
+
+    xx, yy = np.meshgrid(np.arange(dim_pp) - dim_pp // 2, np.arange(dim_pp) - dim_pp // 2)
+
+    if center_pos.lower() == 'b':
+        xx = xx + 1 / 2 - shift[0]
+        yy = yy + 1 / 2 - shift[1]
+    elif center_pos.lower() == 'p':
+        xx = xx - shift[0]
+        yy = yy - shift[1]
+    else:
+        raise Exception("center_pos can only be 'p' or 'b'")
```

Comment:
It might be better to use `ValueError()` here instead of a bare `Exceptoin()`.

Reply:
yes